clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/evaluation/TriggerValue.kt

```diff
@@ -0,0 +1,80 @@
+package com.clevertap.android.sdk.inapp.evaluation
+
+import com.clevertap.android.sdk.variables.JsonUtil
+import org.json.JSONArray
+
+/**
+ * The `TriggerValue` class represents a value used in trigger conditions for in-app messages.
+ * It can encapsulate different types of values, including strings, numbers, and lists.
+ *
+ * @param value The initial value to be encapsulated (default is null).
+ * @param listValue The initial list value to be encapsulated (default is null).
+ */
+class TriggerValue(val value: Any? = null, private var listValue: List<Any?>? = null) {
+
+    private var stringValue: String? = null
+    private var stringValueCleaned: String? = null
+    private var listValueWithCleanedStringIfPresent: List<Any?>? = null
+    private var numberValue: Number? = null
```

Comment:
Is there a need to use  kotlin Number here? for all practical purposes int or long is usually sufficient.

Reply:
tech debt